### ISSUE TITLE
Allow apostrophes in the username

### ISSUE THREAD

**Issue by mdusher** [owncloud/core#20597]
We are using email addresses as the username (inserted using user_saml) and running into issues when a user has a `'` in their email address (which is valid by the RFC5322 standard) because Owncloud doesn't allow the `'` character in a username. 

I've modified the regex in our `core/lib/private/user/manager.php#L257` to allow it and have not yet run into any problems with any of the features we are using (the sync client and apps seem to be working as normal).

### Steps to reproduce
1. Create a user with a `'` in the username

### Expected behaviour
Allow apostrophes in the username

### Actual behaviour
Doesn't allow apostrophes in the username

### Server configuration
**Operating system**: RHEL6
**Web server:** Apache 2.2
**Database:** MariaDB
**PHP version:** 5.6
**ownCloud version:** 8.1.1
**List of activated apps:**

```
 - activity: 2.0.2
 - direct_menu: 0.3
 - external: 1.2
 - files: 1.1.10
 - files_pdfviewer: 0.7
 - files_sharing: 0.6.2
 - files_texteditor: 0.4
 - files_trashbin: 0.6.3
 - files_versions: 1.0.6
 - files_videoviewer: 0.1.3
 - firstrunwizard: 1.1
 - impersonate: 0.0.1
 - provisioning_api: 0.2
 - user_saml: 0.4
```

**Are you using an external user-backend, if yes which one:** No.

**Comment by MorrisJobke** [comment]
cc @karlitschek @DeepDiver1975 @PVince81 regarding usernames in owncloud

**Comment by PVince81** [comment]
cc @blizzz

**Comment by blizzz** [comment]
> I've modified the regex in our core/lib/private/user/manager.php#L257 to allow it and have not yet run into any problems with any of the features we are using (the sync client and apps seem to be working as normal).

@mdusher Would you mind to open a pull request?

**Comment by DeepDiver1975** [comment]
@LukasReschke will this work out with basic auth? THX

**Comment by PVince81** [comment]
PR here https://github.com/owncloud/core/pull/20625

**Comment by LukasReschke** [comment]
> @LukasReschke will this work out with basic auth? THX

Basic Auth: Yes. That's the reason that the encoding is happening using base64. Since this is a ASCII character we also won't get into a encoding mess.
WebDAV: It probably depends, see http://sabre.io/dav/character-encoding/

**Comment by MorrisJobke** [comment]
Fixed with #21806
